### ISSUE TITLE
Unable to bound breakpoint to code under node_modules

### ISSUE THREAD

**Issue by atian25** [Microsoft/vscode#51588]
Issue Type: <b>Bug</b>

**launch.json:**

```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "name": "Debug",
      "type": "node",
      "request": "launch",
      "cwd": "${workspaceRoot}",
      "runtimeExecutable": "npm",
      "runtimeArgs": [
        "run",
        "debug",
        "--",
        "--inspect-brk"
      ],
      "console": "integratedTerminal",
      "protocol": "auto",
      "restart": true,
      "port": 9229,
      "autoAttachChildProcesses": true
    }
  ]
}
```

**package.json:**
```json
{
  "name": "vscode-debug-brk",
  "version": "1.0.0",
  "description": "",
  "dependencies": {
    "is-type-of": "^1.2.0"
  },
  "devDependencies": {},
  "engines": {
    "node": ">=8.0.0"
  },
  "scripts": {
    "debug": "node --inspect-brk index.js"
  }
}
```

**entry:**

```js
// index.js
const is = require('is-type-of');

setInterval(() => {
  console.log(is.string('vscode'));
}, 3000);
```

breakpoint

`node_modules/is-type-of/index.js`

![image](https://user-images.githubusercontent.com/227713/41221064-b58d7672-6d95-11e8-9da6-9f86967f00ca.png)

VS Code version: Code 1.24.0 (6a6e02cef0f2122ee1469765b704faf5d0e0d859, 2018-06-06T17:30:30.113Z)
OS version: Darwin x64 17.5.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-5557U CPU @ 3.10GHz (4 x 3100)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>vpx_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|7, 5, 5|
|Memory (System)|16.00GB (0.92GB free)|
|Process Argv|/Applications/Visual Studio Code.app/Contents/MacOS/Electron|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (30)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-markdownlint|Dav|0.18.0
output-colorizer|IBM|0.1.2
vscode-docker|Pet|0.0.27
Bookmarks|ale|0.19.1
project-manager|ale|0.25.2
copy-syntax|ati|0.5.1
eggjs|ati|0.3.1
vscode-gpm|axe|0.20.0
vscode-custom-css|be5|2.7.1
transformer|dak|1.6.0
vscode-eslint|dba|1.4.12
vscode-npm-source|dku|1.0.0
gitlens|eam|8.3.5
tslint|eg2|1.0.30
code-runner|for|0.9.3
beautify|Hoo|1.3.1
path-autocomplete|ion|1.10.0
search-node-modules|jas|1.3.0
vscode-mocha-sidebar|mat|0.19.1
vsliveshare|ms-|0.3.264
debugger-for-chrome|msj|4.6.0
node-modules-resolve|nau|1.0.2
syncing|non|1.6.2
vetur|oct|0.12.5
gitlink|qez|0.4.3
LiveServer|rit|4.1.0
unittest-navigate|rob|0.0.2
es6-mocha-snippets|spo|0.1.3
code-navigation|vik|0.2.0
JavaScriptSnippets|xab|1.6.0

</details>

**Comment by roblourens** [comment]
Works for me. 

![image](https://user-images.githubusercontent.com/323878/41329691-fb4082ba-6e83-11e8-9666-4583f7f9add8.png)

How did you open the file?

Please set `"trace": true` in your launch config, run it again, and upload the log here.

**Comment by atian25** [comment]
@roblourens sure, see the snapshot above, the file is open.

where is the log will output ?

**Comment by atian25** [comment]
https://gist.github.com/atian25/79987c87cf983f088349d2a1169870ee

**Comment by roblourens** [comment]
For some reason the file is loaded from this path: `/Users/tz/Workspaces/tmp/vscode-debug-brk/node_modules/_is-type-of@1.2.0@is-type-of/index.js`, does that path actually exist on disk for you?

It doesn't do this for me, also on Mac with the same version of Node, weird.

**Comment by atian25** [comment]
@roblourens yes, it's actually exist, I think I found the different, we are using https://github.com/cnpm/npminstall something like `pnpm` to install the `node_modules`, and it will use soft link. 

when I fallback to `npm` and everything is ok.

but, please, anyway to support this?

**Comment by weinand** [comment]
@atian25 does it help if you add the node.js option `--preserve-symlinks`?

**Comment by atian25** [comment]
~~let me try.~~

yes it works
